modules/sdk-coin-eth/src/eth.ts

```diff
@@ -942,6 +1003,239 @@ export class Eth extends BaseCoin {
+        },
+      }),
+      MPC.signCombine({
+        gShare: signConvertS21.gShare as ECDSA.GShare,
+        signIndex: {
+          i: (signConvertS21.muShare as ECDSA.MUShare).i,
+          j: (signConvertS21.muShare as ECDSA.MUShare).j,
+        },
+      }),
+    ];
+
+    const MESSAGE = Buffer.from(txHex);
+
+    // const hashGenerator = (hashType?: string): Hash | undefined => {
+    //   return hashType === 'keccak256' ? createKeccakHash('keccak256') : undefined;
+    // };
+
+    const [signA, signB] = [
+      MPC.sign(MESSAGE, signCombineOne.oShare, signCombineTwo.dShare),
+      MPC.sign(MESSAGE, signCombineTwo.oShare, signCombineOne.dShare),
+    ];
+
+    const signature = MPC.constructSignature([signA, signB]);
+
+    return signature;
+  }
+
+  async recoverTSS(params: RecoverOptions): Promise<RecoveryInfo | OfflineVaultTxInfo> {
+    // this.validateRecoveryParams(params);
+    const isUnsignedSweep = getIsUnsignedSweep(params);
+
+    // Clean up whitespace from entered values
+    const userKey = params.userKey.replace(/\s/g, '');
+    const backupKey = params.backupKey.replace(/\s/g, '');
+
+    // Set new eth tx fees (using default config values from platform)
+    const gasLimit = new optionalDeps.ethUtil.BN(this.setGasLimit(params.gasLimit));
+    const gasPrice = params.eip1559
+      ? new optionalDeps.ethUtil.BN(params.eip1559.maxFeePerGas)
+      : new optionalDeps.ethUtil.BN(this.setGasPrice(params.gasPrice));
+
+    let backupKeyAddress;
+    let userKeyCombined;
+    let backupKeyCombined;
+
+    if (isUnsignedSweep) {
+      const backupKeyPair = new KeyPairLib({ pub: backupKey });
+      backupKeyAddress = backupKeyPair.getAddress();
+    } else {
+      let backupPrv;
+      let userPrv;
+      try {
+        backupPrv = this.bitgo.decrypt({
+          input: backupKey,
+          password: params.walletPassphrase,
+        });
+        userPrv = this.bitgo.decrypt({
+          input: userKey,
+          password: params.walletPassphrase,
+        });
+      } catch (e) {
+        throw new Error(`Error decrypting backup keychain: ${e.message}`);
+      }
+
+      const userSigningMaterial = JSON.parse(userPrv) as ECDSAMethodTypes.SigningMaterial;
+      const backupSigningMaterial = JSON.parse(backupPrv) as ECDSAMethodTypes.SigningMaterial;
+
+      if (!userSigningMaterial.backupNShare) {
+        throw new Error('Invalid user key - missing backupNShare');
+      }
+
+      if (!backupSigningMaterial.userNShare) {
+        throw new Error('Invalid backup key - missing userNShare');
+      }
+
+      const MPC = new Ecdsa();
+
+      userKeyCombined = MPC.keyCombine(userSigningMaterial.pShare, [
+        userSigningMaterial.bitgoNShare,
+        userSigningMaterial.backupNShare,
+      ]);
+      backupKeyCombined = MPC.keyCombine(backupSigningMaterial.pShare, [
+        backupSigningMaterial.bitgoNShare,
+        backupSigningMaterial.userNShare,
+      ]);
+
```

Comment:
validate userKeyComined and backupKeyCombined agree on public key

`if (userKeyCombined.xShare.y !== backupKeyCombined.xShare.y || `userKeyCombined.xShare.chaincode !== backupKeyCombined.xShare.chaincode) {
    throw new Error('Common keychains do not match');
}